string-manipulation-2/src/capitalize-words.js

```diff
@@ -0,0 +1,27 @@
+/* exported capitalizeWords */
+function capitalizeWords(string) {
+  const wordArray = string.split(' ');
+  let result = '';
+  // iterate over words
+  for (let wordIndex = 0; wordIndex < wordArray.length; wordIndex++) {
+    // iterate over characters
+    for (
+      let charIndex = 0;
+      charIndex < wordArray[wordIndex].length;
+      charIndex++
+    ) {
+      // if first character, capitalize, else lowercase
+      if (charIndex === 0) {
+        result += wordArray[wordIndex].charAt(charIndex).toUpperCase();
+      } else {
+        result += wordArray[wordIndex].charAt(charIndex).toLowerCase();
+      }
+    }
+
+    // add a space if the word is not the last word
+    if (wordIndex !== wordArray.length - 1) {
+      result += ' ';
+    }
+  }
+  return result;
+}
```

Comment:
You want to try and avoid nested loops as much as you can, because they increase the overall complexity of our code. Here's a solution that only uses 1 loop
```
function capitalizeWords(string) {
  const words = string.split(' ');
  for (let i = 0; i < words.length; i++) {
    const word = words[i];
    const capitalized = word[0].toUpperCase() + word.slice(1).toLowerCase();
    words[i] = capitalized;
  }
  return words.join(' ');
}
```